inspirehep/modules/tools/authortool.py

```diff
@@ -0,0 +1,297 @@
+        if aff_type:
+            aff_pattern = line_pattern_line[aff_type]
+        else:
+            raise ValueError('Cannot identify type of affiliations', first_char)
+
+    return aff_pattern
+
+
+def parse_affiliations(text):
+    """
+    Determine how affiliations are formatted.
+    Return hash of id:affiliation
+
+    Allowed formats:
+    don't mix letters and numbers, lower-case letters only
+
+    1
+    CERN, Switzerland
+    2
+    DESY,
+    Germany
+
+
+    1 CERN, Switzerland
+    2DESY, Germany
+
+    a  CERN, Switzerland
+    bb DESY, Germany
+
+    *
+    CERN, Switzerland
+    #
+    DESY, Germany
+    """
+
+    affiliations = {}
+    aff_pattern = determine_aff_type(text)
+
+    aff_id = None
+    this_aff = []
+    for line in text.split('\n'):
+        line = line.strip()
+        get_affiliation = aff_pattern.search(line)
+        if get_affiliation:
+            if len(get_affiliation.groups()) == 2:
+                affiliations[get_affiliation.group(1)] = \
+                    get_affiliation.group(2).strip()
+            else:
+                if aff_id and this_aff:
+                    affiliations[aff_id] = ' '.join(this_aff).strip()
+                    aff_id = None
+                    this_aff = []
+                aff_id = get_affiliation.group(1)
+        elif aff_id:
+            this_aff.append(line)
+        elif line:
+            raise ValueError('Something is wrong with the affiliation list')
+    if aff_id and this_aff:
+        affiliations[aff_id] = ' '.join(this_aff).strip()
+
+    return affiliations
+
+
+def create_authors(text):
+    """
+    Split text in (useful) blocks, sepatated by empty lines.
+    1 block: no affiliations
+    2 blocks: authors and affiliations
+    more blocks: authors grouped by affiliation (not implemented yet)
+
+    Returns: 
+    List of tuples: (author_fullname, [author_affiliations])
+    String: warning
+    """
+
+    warning = ""
+    authors = []
+    
+    if not text:
+        return  authors, warning    
+
+    if not isinstance(text, six.text_type):
+        text = text.decode('utf-8')
+    text = text.replace('\r', '')  # Input from the form contains unwanted \r's
+    text = re_hyphens.sub('-', text)
+
+    empty_blocks = []
+    text_blocks = re_emptyline.split(text)
+    for num, block in enumerate(text_blocks):
+        if not re.search(r'\w', block):
+            empty_blocks.append(num)
+    empty_blocks.reverse()
+    for num in empty_blocks:
+        text_blocks.pop(num)
+
+
+    if len(text_blocks) == 0:
+        authors = []
+    elif len(text_blocks) == 1:
+        authors, text = parse_authors(text_blocks[0], {})
+        warning += text
+    elif len(text_blocks) == 2:
+        affiliations = parse_affiliations(text_blocks[1])
+        authors, text = parse_authors(text_blocks[0], affiliations)
+        warning += text
+    else:
+#        authors = parse_blocks(text_blocks)
+        authors = []
+
+    return  authors, warning
```

Comment:
Minor thing: dict-like returns like the one I proposed in https://github.com/inspirehep/inspire-next/pull/3054#discussion_r156929280 are more extensible than returning tuples, because you can always add keys to a dictionary without breaking a client, and that's not the case with sequence types.